ddtrace/appsec/utils.py

```diff
@@ -13,13 +22,80 @@ def _appsec_rc_features_is_enabled():
 
 def _appsec_rc_capabilities():
     # type: () -> str
-    """return the bit of the composed capabilities in base64
+    r"""return the bit representation of the composed capabilities in base64
     bit 0: Reserved
-    bit 1: ASM Activation
+    bit 1: ASM 1-click Activation
     bit 2: ASM Ip blocking
 
-    TODO: refactor to compose the string and encode it
+    Int Number  -> binary number    -> bytes representation -> base64 representation
+    ASM Activation:
+    2           -> 10               -> b'\x02'              -> "Ag=="
+    ASM Ip blocking:
+    4           -> 100              -> b'\x04'              -> "BA=="
+    ASM Activation and ASM Ip blocking:
+    6           -> 110              -> b'\x06'              -> "Bg=="
+    ...
+    256         -> 100000000        -> b'\x01\x00'          -> b'AQA='
     """
+    value = 0b0
+
     if _appsec_rc_features_is_enabled():
-        return "Ag=="
-    return ""
+        value |= 1 << 1
+        value |= 1 << 2
+
+    result = ""
+    if sys.version_info.major < 3:
+        bytes_res = to_bytes_py2(value, (value.bit_length() + 7) // 8, "big")
+        result = base64.b64encode(bytes_res)  # type: ignore[assignment, arg-type]
+    else:
+        result = str(base64.b64encode(value.to_bytes((value.bit_length() + 7) // 8, "big")), encoding="utf-8")
+
+    return result
+
+
+_HTML_BLOCKED_TEMPLATE_CACHE = None  # type: Optional[str]
+_JSON_BLOCKED_TEMPLATE_CACHE = None  # type: Optional[str]
+
+
+def _get_blocked_template(accept_header_value):
+    # type: (str) -> str
+
+    global _HTML_BLOCKED_TEMPLATE_CACHE
+    global _JSON_BLOCKED_TEMPLATE_CACHE
+
+    need_html_template = False
+
+    if accept_header_value and "text/html" in accept_header_value.lower():
+        need_html_template = True
+
+    if need_html_template and _HTML_BLOCKED_TEMPLATE_CACHE:
+        return _HTML_BLOCKED_TEMPLATE_CACHE
+
+    if not need_html_template and _JSON_BLOCKED_TEMPLATE_CACHE:
+        return _JSON_BLOCKED_TEMPLATE_CACHE
+
+    if need_html_template:
+        template_path = os.getenv("DD_APPSEC_HTTP_BLOCKED_TEMPLATE_HTML")
+    else:
+        template_path = os.getenv("DD_APPSEC_HTTP_BLOCKED_TEMPLATE_JSON")
+
+    if template_path and os.path.exists(template_path) and os.path.isfile(template_path):
```

Comment:
should we warn if there is a template path given, but the file does not exist?

I wonder if it is worth just removing the exists/isfile check, since `open()` will raise that exception for us?

not sure I love my suggestion, but feels like there is a feedback point missing for when a path is given, but the path does not exist.

Reply:
Removed the isfile/exists checks. The exception potentially raised by open (could not load custom template...) should be enough hint for the user.